### ISSUE TITLE
synapse.swarm.syntax should not raise core exceptions

### ISSUE THREAD

**Issue by aheinz-fe** [vertexproject/synapse#82]
The parse methods raise mostly Exception, forcing my enclosing exception handler to be overly generic. How about consistently raising [SyntaxError](https://docs.python.org/2/library/exceptions.html#exceptions.SyntaxError) with a helper function like:

```
def raise_syntax_error(message, offset, text):
    exc = SyntaxError(message)
    exc.offset = offset
    exc.text = text
    raise exc
```

**Comment by williballenthin** [comment]
rather than using a helper to raise the exception, why not subclass `SyntaxError`:

    class SwarmSyntaxError(SyntaxError):
        def __init__(self, message, offset, text):
            super(SwarmSyntaxError, self).__init__(message)
            self.offset = offset
            self.text = text

then we can do:

    raise SwarmSyntaxError(...)

**Comment by aheinz-fe** [comment]
I'm fine with either.

**Comment by williballenthin** [comment]
addressed in #93

**Comment by invisig0th** [comment]
@williballenthin thanks for #93
